core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/configuration/security/jwt/JWTAuthenticationFilter.java

```diff
@@ -23,14 +25,26 @@ public JWTAuthenticationFilter(JWTAuthenticationConverter jwtAuthenticationConve
     }
 
     @Override
-    protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain) throws ServletException, IOException {
-        Authentication authentication = jwtAuthenticationConverter.convert(httpServletRequest);
-        if (authentication == null) {
+    public void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain) throws IOException, ServletException {
+        try {
+            Authentication authentication = jwtAuthenticationConverter.convert(httpServletRequest);
+            SecurityContextHolder.getContext().setAuthentication(authentication);
             filterChain.doFilter(httpServletRequest, httpServletResponse);
-            return;
+        } catch (JWTVerificationException e) {
+            setErrorResponse(HttpStatus.UNAUTHORIZED, httpServletResponse, e);
+            e.printStackTrace();
         }
-        SecurityContextHolder.getContext().setAuthentication(authentication);
-        filterChain.doFilter(httpServletRequest, httpServletResponse);
     }
 
+    public void setErrorResponse(HttpStatus status, HttpServletResponse response, Throwable ex) {
+        response.setStatus(status.value());
+        response.setContentType("application/json");
+        ApiError apiError = new ApiError(status, ex);
+        try {
+            String json = apiError.convertToJson();
+            response.getWriter().write(json);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
let the method setErrorResponse throw the IOException instead of just printing the stackTrace